.github/workflows/test.yml

```diff
@@ -77,32 +77,34 @@ jobs:
         if [ -f ${CI_IMAGE_CACHE}${CI_IMAGE_MASTER_TAR}.gz ]; then
             cp ${CI_IMAGE_CACHE}/${CI_IMAGE_MASTER_TAR}.gz ${CI_IMAGE_MASTER_TAR}.gz
             gunzip ${CI_IMAGE_MASTER_TAR}.gz
-            echo "::set-output name=MASTER_IMAGE_RESTORED_FROM_CACHE::true"
+            echo "MASTER_IMAGE_RESTORED_FROM_CACHE=true" >> "$GITHUB_OUTPUT"
+            echo "MASTER_IMAGE_RESTORED=true" >> "$GITHUB_OUTPUT"
             exit 0
         fi
 
         if docker pull ghcr.io/ovn-org/ovn-kubernetes/ovn-kube-f:master; then
             docker tag ghcr.io/ovn-org/ovn-kubernetes/ovn-kube-f:master ovn-daemonset-f:dev
 
-            echo "::set-output name=MASTER_IMAGE_RESTORED_FROM_GHCR::true"
+            echo "MASTER_IMAGE_RESTORED=true" >> "$GITHUB_OUTPUT"
             exit 0
         fi
     # only run the following steps if the master image was not found in the cache
+    - name: Check out code into the Go module directory - from master branch
+      if: steps.is_master_image_build_needed.outputs.MASTER_IMAGE_RESTORED != 'true' && success()
+      uses: actions/checkout@v4
+      with:
+        ref: master
+
     - name: Set up Go
       if: steps.is_master_image_build_needed.outputs.MASTER_IMAGE_RESTORED != 'true' && success()
-      uses: actions/setup-go@v3
+      uses: actions/setup-go@v5
       with:
         go-version: ${{ env.GO_VERSION }}
+        cache-dependency-path: "**/*.sum"
       id: go
 
-    - name: Check out code into the Go module directory - from master branch
-      if: steps.is_master_image_build_needed.outputs.MASTER_IMAGE_RESTORED_FROM_GHCR != 'true' && steps.is_master_image_build_needed.outputs.MASTER_IMAGE_RESTORED_FROM_CACHE != 'true' && success()
-      uses: actions/checkout@v3
-      with:
-        ref: master
-
     - name: Build - from master branch
-      if: steps.is_master_image_build_needed.outputs.MASTER_IMAGE_RESTORED_FROM_GHCR != 'true' && steps.is_master_image_build_needed.outputs.MASTER_IMAGE_RESTORED_FROM_CACHE != 'true' && success()
+      if: steps.is_master_image_build_needed.outputs.MASTER_IMAGE_RESTORED != 'true' && success()
```

Comment:
we don't want to use the CACHE logic anymore? I can see:
```
Additionally fixed the logic around MASTER_IMAGE_RESTORED as it was causing warnings with caching and removed Build k8s periodic job as it doesn't actually do anything useful, AFAICT, as well as Install KIND step.
```
this in PR message but checking in as to why we want to remove that..

Reply:
We're not removing the caching.  Maybe the sentence is a little confusing.  I'm just replacing two separate variables with one.  `MASTER_IMAGE_RESTORED` will be true in case we restored from either the CACHE or GHCR.  This way we can simplify the `if` condition.

The main change here is actually the fact that we are not changing `Set up Go` step.  The `MASTER_IMAGE_RESTORED` doesn't exist on current ovn-k master, but `Set up Go` step is using this undefined variable.  So, the `if` condition is always `true` for it and we're setting up Go every time even if everything is cached.  I introduced the `MASTER_IMAGE_RESTORED` variable, so the `if` condition in `Set up Go` can become `false` when cached.

Does that make sense?

Reply:
```
 I'm just replacing two separate variables with one. MASTER_IMAGE_RESTORED will be true in case we restored from either the CACHE or GHCR. This way we can simplify the if condition.
```
ack gotcha.

Reply:
the commit message is super detailed and outlining exactly what is happening, my bad should have read that first.